framework/include/utils/MooseUtils.h

```diff
@@ -383,6 +385,51 @@ tokenizeAndConvert(const std::string & str,
   }
   return true;
 }
+
+/**
+ * Returns the default execute_on MultiMooseEnum.
+ * @param default_names Space separated list to set the default execute flags.
+ */
+MultiMooseEnum createExecuteOnEnum(const std::set<ExecFlagType> & set_flags = {},
+                                   const std::set<ExecFlagType> & add_flags = {},
+                                   const std::set<ExecFlagType> & remove_flags = {});
+
+///@
+/**
+ * Add/removes/sets the given execute flags to the InputParameters or MultiMooseEnum object.
+ * @param params/exec_enum The InputParameters or MultiMooseEnum object to modify.
+ * @param flags A set of ExecFlagType to append as possible values.
+ *
+ * The add/remove methods also automatically update the doc string for the "execute_on" parameter.
+ */
+void addExecuteOnFlags(InputParameters & params, const std::set<ExecFlagType> & flags);
+void addExecuteOnFlags(MultiMooseEnum & exec_enum, const std::set<ExecFlagType> & flags);
+void removeExecuteOnFlags(InputParameters & params, const std::set<ExecFlagType> & flags);
+void removeExecuteOnFlags(MultiMooseEnum & exec_enum, const std::set<ExecFlagType> & flags);
+void setExecuteOnFlags(InputParameters & params, const std::set<ExecFlagType> & flags);
+void setExecuteOnFlags(MultiMooseEnum & exec_enum, const std::set<ExecFlagType> & flags);
```

Comment:
I explained this a bit somewhere. I would of liked to created a class inheriting from MultiMooseEnum, but all the plumbing necessary to add the class to InputParameters was a lot more code than have some helper functions. 

I also believe from a user perspective it is easier to use a MultiMooseEnum in the InputParameters rather than add another special type that users will not be familiar with.